src/tests/json-stringify.ts

```diff
@@ -0,0 +1,29 @@
+import { doNotExecute, Equal, Expect } from "./utils";
+
+doNotExecute(() => {
+  const result = JSON.stringify(undefined);
+  type tests = [Expect<Equal<typeof result, undefined>>];
+});
+
+doNotExecute(() => {
+  const result = JSON.stringify(5 as undefined | number);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
+});
+
+doNotExecute(() => {
+  // If the type of the incoming value is `any` we can't assume anything about
+  // it so return the broadest possible type
+  const result = JSON.stringify(undefined as any);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
+});
+
+doNotExecute(() => {
+  const result = JSON.stringify(undefined as unknown);
+  type tests = [Expect<Equal<typeof result, string | undefined>>];
+});
+
+doNotExecute(() => {
+  const value: null | string | number | boolean | object | Array<any> = null;
```

Comment:
despite you put a type for a value variable, TS narrows it to `null` ([see here](https://tsplay.dev/mpjMpw)). update it with `as` to get expected type

```suggestion
  const value = null as (null | string | number | boolean | object | Array<any>);
```

Reply:
Done.